Entrega_Proyecto_Final_1/src/DAO/mongo/models/ticket.models.js

```diff
@@ -0,0 +1,32 @@
+import mongoose from 'mongoose';
+
+const TicketSchema = new mongoose.Schema({
+  _id: {
+    type: mongoose.SchemaTypes.ObjectId,
+    default: new mongoose.Types.ObjectId(),
+  },
+  amount: {
+    type: Number,
+    default: "No se establecio el precio"
+  },
+  purchaser: { 
+    type: String,
+    required: true,
+  },
+  cartId: {
+    type: mongoose.SchemaTypes.ObjectId,
+    ref: 'carts',
+  },
+  user: {
+    type: mongoose.SchemaTypes.ObjectId,
+    ref: 'users',
+  },
+  status: {
+    type: String,
+    default: "abierto" 
+  },
+});
+
```

Comment:
Te faltaria el atributo code y purchase_datetime

Reply:
Pero lo deje generado aca mira 
  createTicket = async (user, cartId) => {
    const amount = 100;


    const code = crypto.randomBytes(4).toString('hex');
  
    const ticket = {
      code: code,
      purchase_datetime: new Date(),
      amount: amount,
      purchaser: user ? user.email : 'email@example.com', 
      cartId: cartId,
      user: user ? user._id : null, 
    };
  
    return await this.dao.create(ticket);  
  };

Reply:
como que lo genere directamente del services  , esta mal?

Reply:
sisi, esta perfecto pero te faltaria agregar los atributos al modelo, ya que si no se especifica el atributo en el modelo no se guarda en la db

Reply:
Sabes, estuve hablando con los demas tutores y coincidimos que en vez del cartId tiene que haber un array, podes directamente incrustar los productos que se compraron nomas al crear un ticket. Algo asi deberia quedar:

{
  "code": "asda",
  "products": [
    {
      "productId": {
        "$oid": "64d2af68d4e4184c004f76a0"
      },
      "quantity": 3
    }
  ],
  "amount": 389.97,
  "purchaser": "santi@example.com",
  "purchase_datetime":  "2023-10-02T04:14:18.796Z"
}